public/scripts/index.js

```diff
@@ -1,138 +1,162 @@
-    }
-
-    if (currentFrame === undefined){
-        alert('you must draw a grid')
-    }
-    else if (numValue === 0){
-        alert('You must select a preset');
-    }
-    else if (currentFrame.length < 17){
-        alert('Grid is too small. The minimum size is 17x17');
-    }    
+function drawPreset () {
+  let Nwid = (Math.ceil(window.innerWidth / 8) * 8)
+  let Nhei = (Math.ceil(window.innerHeight / 8) * 8)
+  let wid = Nwid - 32
+  let hei = Nhei - 32
+  const arrH = hei / cellSize
+  const arrW = wid / cellSize
+  currentFrame = makeMatrix(arrH, arrW)
+  const value = getPreset()
+  const numValue = parseInt(value)
+  if (numValue === 1 && currentFrame.length >= 17 && currentFrame[16][16] !== undefined) {
+    console.log(currentFrame)
+    clear()
+    drawToad(currentFrame, context, cellSize)
+  }
+  if (numValue === 2 && currentFrame.length >= 17 && currentFrame[16][16] !== undefined) {
+    clear()
+    drawAcorn(currentFrame, context, cellSize)
+  }
+  if (numValue === 3 && currentFrame.length >= 17 && currentFrame[16][16] !== undefined) {
+    clear()
+    drawLWSS(currentFrame, context, cellSize)
+  }
+
+  if (currentFrame === undefined) {
+    alert('you must draw a grid')
+  } else if (numValue === 0) {
+    alert('You must select a preset')
+  } else if (currentFrame.length < 17) {
+    alert('Grid is too small. The minimum size is 17x17')
+  }
 }
 
-function clear(){
-    let rows = document.getElementById("rows").value;
-    let cols = document.getElementById("columns").value;
-    currentFrame = makeMatrix(rows, cols);
-    intermediateFrame = makeMatrix(rows, cols);
-    nextFrame = makeMatrix(rows, cols);
-    draw(currentFrame, context, cellSize);
+function clear () {
+  let Nwid = (Math.ceil(window.innerWidth / 8) * 8)
+  let Nhei = (Math.ceil(window.innerHeight / 8) * 8)
+  let wid = Nwid - 32
+  let hei = Nhei - 32
+  const rows = hei / cellSize
+  const cols = wid / cellSize
+  currentFrame = makeMatrix(rows, cols)
+  intermediateFrame = makeMatrix(rows, cols)
+  nextFrame = makeMatrix(rows, cols)
+  draw(currentFrame, context, cellSize)
 }
-function animate(){
-    let checkedVal = document.getElementById('animate').checked;
-    console.log(checkedVal);
-    if(checkedVal === true){
-        setTimeout(function(){ simulate(); }, 500);
-        setTimeout(function(){ animate(); }, 500);
-    }
-    else{
-        console.log('NOT ANIMATING');
-    }
+function animate () {
+  const checkedVal = document.getElementById('animate').checked
+  console.log(checkedVal)
+  if (checkedVal === true) {
+    setTimeout(function () { simulate() }, 500)
+    setTimeout(function () { animate() }, 500)
+  } else {
+    console.log('NOT ANIMATING')
+  }
 }
 
-function alert(){
-    console.log('ALERT');
-}
+function resize () {
+  let Nwid = (Math.ceil(window.innerWidth / 8) * 8)
+  let Nhei = (Math.ceil(window.innerHeight / 8) * 8)
+  let wid = Nwid - 32
+  let hei = Nhei - 32
 
+  canvas.height = hei
```

Comment:
The resize func is not testable because it contains references to window and canvas which are DOM elements. Refactor it so that it gets the required stuff as parameters. You should change the signature of your func to smth like this `function resize(winH, winW)`and in the caller func you would do this `resize(window.innerHeight, window.innerWidth)` Also instead of setting the height and width of the canvas directly you should just return the new values as a point object for the caller to handle

Reply:
After making it testable you can put it in the engine.js and and write a test for it